docs/design-docs/overview.md

```diff
@@ -72,6 +72,7 @@ Here're the details about work directory structure for SuperBench Runner.
         │       └── sb-exec.log               # collected SuperBench Executor log
         ├── sb-run.log                        # SuperBench Runner log
         ├── sb.config.yaml                    # SuperBench configuration snapshot
+        ├── pattern_config                    # generated pattern config file for diagnosis
```

Comment:
Is it optional?

Reply:
It might be ambiguous if using optional, cause its generation is implied under mpi pattern which is not optional. How about `generated host groups config under specified patterns for diagnosis`

Reply:
I don't think so. For monitor, it's also optional because you can close monitor. it's the same for pattern. Also, if it's for diagnosis, why do we must have it?

Reply:
Add a a bool option `mpi_pattern` under pattern for user, so that they can use this option to get config file for diagnosis if they want,